src/lib/optimizer/join_ordering/join_graph_builder.cpp

```diff
@@ -32,6 +40,11 @@ std::optional<JoinGraph> JoinGraphBuilder::operator()(const std::shared_ptr<Abst
   // A single vertex without predicates is not considered a JoinGraph
   if (_vertices.size() <= 1u && edges.empty()) return std::nullopt;
 
+  for (auto& vertex : _vertices) {
+    DebugAssert(vertex->type == LQPNodeType::Root, "Expected root of vertex to be LogicalPlanRootNode");
+    vertex = vertex->left_input();
```

Comment:
We should probably untie the root node first so that the outputs of `vertex` remain correct.